### PR TITLE
1645 move organelle center of mass to cell origin

### LINKED ISSUE #1645: Make cells rotate around their center of mass (fractional offset)
instead of always having the hex at 0,0 be the center of the cell in the stage.
Instead could compute average position of all hexes and use that as offset all the organelles when rendering and in physics.
This will prevent weirdness if the player doesn't build their cell around the middle of the editor grid.

closes #1678

### LINKED ISSUE #1678: Your only part can be out of origin
You can "strech" your being by "moving" the origin

**Describe the bug**
When the only structure is **not** the origin, weird things can happen

**To Reproduce**
1. Place a line
grafik
2. Remove the line, starting with the oldest part
3. Repeat until your only part is far away from the origin

**Expected behavior**
The editor and ingame-origin should be recalculated when clicking confirm. See #1668

**Screenshots**
grafik
grafik
When abusing this to the extreme, your cell can get invisible

### PR DESCRIPTION
**Pull request by FuJa0815**
A simpler version of #1668

Closes #1678

### CODE REVIEW

**Comment by FuJa0815** [comment]
Oh, I almost forgot to thanks @Huitsi for the Center of mass code!

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -463,10 +463,11 @@ public void OnFinishEditing()
 
         // It is easiest to just replace all
         editedSpecies.Organelles.Clear();
-
+        var centerOfMass = editedMicrobeOrganelles.CenterOfMass;
```
Comment (Kasterisk) [inline src/microbe_stage/editor/MicrobeEditor.cs:466]: I feel it would be nicer to have empty lines before and after this

**Review by Kasterisk — COMMENTED** [review]
Functionally seem to be good, collision is fine as far as I can tell

File: src/microbe_stage/OrganelleLayout.cs [hunk 2]
```diff
@@ -34,6 +35,25 @@ public OrganelleLayout()
 
     public bool IsReadOnly => false;
 
+    /// <summary>
+    /// The center of mass of the contained organelles.
```
Comment (hhyyrylainen) [inline src/microbe_stage/OrganelleLayout.cs:39]: XML contents should be intended by 2 spaces.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 3]
```diff
@@ -463,10 +463,11 @@ public void OnFinishEditing()
...
         foreach (var organelle in editedMicrobeOrganelles.Organelles)
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:467]: I'd like a blank line before this foreach

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 4]
```diff
@@ -463,10 +463,11 @@ public void OnFinishEditing()
...
         {
             var organelleToAdd = (OrganelleTemplate)organelle.Clone();
+            organelleToAdd.Position -= centerOfMass;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:470]: please add a comment here to explain why the center of mass is subtracted here.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Functionality wise this looks good. A few code cleanup comments.
Also this doesn't fully solve #1645 because for that fractional positioning is wanted.

File: src/microbe_stage/editor/MicrobeEditor.cs (hunk 3 above)
Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs:468]: You added the blank line Kasterisk wanted, I'd still like to separate the foreach block from the assignment on the previous line.

Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs:468]: That my previous comment was the review I wanted to do

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/microbe_stage/editor/MicrobeEditor.cs (hunk 3 above)
Comment (FuJa0815) [reply src/microbe_stage/editor/MicrobeEditor.cs:468]: Sorry, didn't read accurately enough. I'll do that tomorrow

**Review by hhyyrylainen — APPROVED** [review]
I did playtest this before and now the style issues are fixed, so I'll assume this still works.

**Pull request merged**